cdc/processor/sinkmanager/mem_quota.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sinkmanager
+
+import "github.com/pingcap/tiflow/cdc/model"
+
+//nolint:varcheck,deadcode
+const defaultMemoryUsage = 10 * 1024 * 1024 // 10MB
+
+type memQuota interface {
+	// TryAcquire try to acquire n bytes of memory.
+	TryAcquire() bool
+	// ForceAcquire force to acquire n bytes of memory.
+	// It will exceed the total memory limit.
+	ForceAcquire()
```

Comment:
Seems we need to comment `there will at most 1 client can acquire memory exceeds the limit`.

Reply:
I guess it's OK to have some workers exceed the limitation. Compared to the old implementation, we exceeded the limits to a manageable degree. Originally it was the number of tables that determined this, but now it is up to n workers.

Reply:
If table count is less than worker count, this implementation is same bad with the old one. I guess it shouldn't be very hard to make it better.

Reply:
> I guess it shouldn't be very hard to make it better.

Now it's a multi-table share, in which case competing workers are coordinated and they can affect each other. For example, what if the slowest kind of table doesn't get the only quota? This would still affect latency.

Reply:
If memory quota full really happens, I think it's more important to avoid OOM instead of take care about latency.

Reply:
Seems even if at most 1 client can acquire memory on quota full, OOM is still possible to happen.

Let's ignore OOM problem here. We can deal with it in another place.